README.md

```diff
@@ -66,85 +66,118 @@ See [parse feature](features/parse.feature) for the details on using the `drafte
-blueprint parsed and serialized as [API
-Elements](http://api-elements.readthedocs.io) in YAML or JSON.
+The `drafter_parse_blueprint_to` function translates a buffered API blueprint into [API
+Elements](http://api-elements.readthedocs.io), serialized into one of its supported serialization formats.
 
 ```c
-int drafter_parse_blueprint_to(const char* source, char ** out, const drafter_options options);
+drafter_error drafter_parse_blueprint_to(
+    const char* source,
+    char** out,
+    const drafter_parse_options* parse_opts,
+    const drafter_serialize_options* serialize_opts);
+);
 ```
 
-```c
-#include <drafter/drafter.h>
+Given a pointer to a UTF-8 encoded c-string,
 
+```c
 const char* blueprint =
   "# My API\n"
   "## GET /message\n"
   "+ Response 200 (text/plain)\n"
   "\n"
   "      Hello World!\n";
+```
 
-drafter_options options;
-options.format = DRAFTER_SERIALIZE_JSON;
-options.sourcemap = true;
+##### Serialized as YAML
 
-char* result = NULL;
-if (drafter_parse_blueprint_to(blueprint, &result, options) == 0) {
-    printf("%s\n", result);
-    free(result);
+Without options, the resulting API Elements is serialized as YAML.
+
+```c
+char* yamlApie = NULL;
+if (drafter_parse_blueprint_to(blueprint, &yamlApie, NULL, NULL) == 0) {
+    printf("%s\n", yamlApie);
 }
+
+free(yamlApie);
 ```
 
-#### Checking the validity of a blueprint
+##### Serialized as JSON
 
-The `drafter_check_blueprint` function allows checking the validity of a
-blueprint. This function will return a `drafter_result` when the blueprint
-produces warnings and/or errors.  With a `drafter_result`, the
-`drafter_serialize` function can be used to serialized the result as [API
-Elements](http://api-elements.readthedocs.io) in YAML or JSON.
+Tweaking `drafter_serialize_options` allows serialization into JSON.
 
+```c
+drafter_serialize_options* options = drafter_init_serialize_options();
+drafter_set_format(options, DRAFTER_SERIALIZE_JSON);
+
+char* jsonApie = NULL;
+if (0 == drafter_parse_blueprint_to(blueprint, &jsonApie, NULL, NULL)) {
+    printf("%s\n", jsonApie);
+}
+
+free(jsonApie);
+drafter_free_options(options);
+```
+
+#### Validate API Blueprint
+
+API Blueprint can be validated via `drafter_check_blueprint`. 
 
 ```c
-drafter_result* drafter_check_blueprint(const char* source);
+drafter_error drafter_check_blueprint(
+    const char* source,
+    drafter_result** res,
+    const drafter_parse_options* parse_opts);
 ```
 
+##### Simple validation
+
+The return value of `drafter_check_blueprint` indicates validation success.
+
 ```c
-#include <drafter/drafter.h>
+drafter_result* result = NULL;
 
-const char* blueprint =
-  "# My API\n"
-  "## GET /message\n"
-  "+ Response 200 (text/plain)\n"
-  "\n"
-  "      Hello World!\n";
+if (0 == drafter_check_blueprint(blueprint, result)) {
+    print("Understood.\n");
+} else {
+    print("Invalid API Blueprint.?\n");
+}
+
+drafter_free_result(result);
```

Comment:
Under what cases do we need to `free(result)`? It's not handled in any.

Reply:
The user reading a README and developing his code upon only it, without looking into the header, is a user that should _always_ call `drafter_free_result`. It's not like it's not a `NOOP` when the user doesn't need the invocation. Also, notice the amount of negations in the last sentence.

Reply:
🤦‍♂ I didn't realise `result` was passed to `drafter_free_result`. Ignore this :)